app/services/notification_service/course_subjects_updated.rb

```diff
@@ -0,0 +1,42 @@
+module NotificationService
+  class CourseSubjectsUpdated
+    include ServicePattern
+
+    def initialize(
+      course:,
+      previous_subject_names:,
+      previous_course_name:
+    )
+      @course = course
+      @previous_subject_names = previous_subject_names
+      @previous_course_name = previous_course_name
+    end
+
+    def call
+      return false unless notify_accredited_body?
+      return false unless course.in_current_cycle?
+
+      users = User.joins(:user_notifications).merge(UserNotification.course_update_notification_requests(course.accredited_body_code))
```

Comment:
you can extract this out to a private method instead of using a local variable

Reply:
Could do but what in way would doing so improve things? I would _extract method_ if:

- doing so made the code more readable (we already have a users variable and it already seems clear that we are pulling users from the db?)
- There was more than one place in the class that was using the same logic. In this case there isn't.